### PR TITLE
thumbs-down to remove ninja stars

### PR DESCRIPTION
**Pull request by gruiz17**
…, write more tests

### CODE REVIEW

File: src/server/services/flags.js [hunk 1]
```diff
@@ -53,7 +53,16 @@ module.exports = {
...
         var starRegex = /\!star|\!ninjastar|\+1|\:thumbsup\:|\:star\:/g;
-        return !!comment.match(starRegex);
+        var unstarRegex = /\!unstar|\-1|\:thumbsdown\:/g;
+        if (comment.match(unstarRegex)) {
+            return 'remove';
+        }
+        else if (comment.match(starRegex) && !comment.match(unstarRegex)) {
+            return 'create'
+        }
+        else {
```
Comment (dfarr) [inline src/server/services/flags.js]: Don't need this else !fix

File: src/server/services/flags.js [hunk 2]
```diff
@@ -53,7 +53,16 @@ module.exports = {
         // input: single conversation comment
         // output: true or false (true to create a ninja star)
         var starRegex = /\!star|\!ninjastar|\+1|\:thumbsup\:|\:star\:/g;
-        return !!comment.match(starRegex);
+        var unstarRegex = /\!unstar|\-1|\:thumbsdown\:/g;
+        if (comment.match(unstarRegex)) {
+            return 'remove';
```
Comment (dfarr) [inline src/server/services/flags.js]: Let's not do it this way. This is an example of tight coupling. The calling function *requires* two functions named 'remove' and 'create'.

Instead let's have a `star` and `unstar` function that simply returns true or false. !resolve

File: src/server/services/flags.js [hunk 3]
```diff
@@ -53,7 +53,13 @@ module.exports = {
...
```
Comment (gruiz17) [inline src/server/services/flags.js]: ok! will fix this

File: src/server/services/flags.js (hunk 1 above)
Comment (gruiz17) [inline src/server/services/flags.js]: !completed

File: src/server/services/flags.js (hunk 3 above)
Comment (gruiz17) [inline src/server/services/flags.js]: !fixed

File: src/server/services/flags.js (hunk 2 above)
Comment (gruiz17) [inline src/server/services/flags.js]: !fixed

**Pull request merged**